src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -264,6 +264,50 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
         }
         break;
     }
+    case SMART_ACTION_RANDOM_SOUND:
+    {
+        ObjectList* targets = GetTargets(e, unit);
+        if (!targets)
+            break;
+
+        uint32 sounds[4];
+        sounds[0] = e.action.randomSound.sound1;
+        sounds[1] = e.action.randomSound.sound2;
+        sounds[2] = e.action.randomSound.sound3;
+        sounds[3] = e.action.randomSound.sound4;
+        uint32 temp[4];
```

Comment:
I think it would be better to use modern-C++ syntax for arrays, have a look:

https://en.cppreference.com/w/cpp/container/array